src/main/java/com/radixdlt/atommodel/Atom.java

```diff
@@ -191,22 +188,22 @@ public void addParticleGroupWith(Particle particle0, Spin spin0, Particle partic
 	}
 
 	// SIGNATURES //
-	public boolean verify(Collection<ECPublicKey> keys) throws PublicKeyException {
-		return this.verify(keys, keys.size());
+	public boolean verify(Collection<ECPublicKey> keys, Hasher hasher) throws PublicKeyException {
```

Comment:
Providing hasher into method which calls methods on same object seems to be a code smell to me. Fine for now but add comment that this should be refactored at some point.

Reply:
what's the alternative?
I originally came up with a static method ( `Atom.verify(atom, keys, hasher)` ) but was requested by Martin to change it

Reply:
I think the general idea would be that at some point we remove hashing/signing/verifying smarts from Atom, and just let it be a POJO with hashing/signing/verifying done as a separate concern.

I think rather than hacking something in here now, it would be best to delay this until we can separate signatures from the data that is being signed.  This seems good enough for now, but I think the refactoring that @talekhinezh  is referring to should be to move signatures out of the (hopefully) POJO data that they are signing.

Reply:
I think that regardless of this refactoring the problem remains. Seems there are just 4 options:
```
signedObj.verify(key, hasher); // instance method
SignedObj.verify(signedObj, key, hasher); // static
SignedObjVerifier.verify(signedObj, key, hasher); // static but in separate class
signedObjVerifier.verify(signedObj, key); // hasher already injected to verifier instance
```

which one would be preferred?

Reply:
The first option which is what you currently have implemented is fine. My comment is more on a refactor for the future.